internal/services/network/validate/web_application_firewall_policy.go

```diff
@@ -49,12 +66,15 @@ var ValidateWebApplicationFirewallPolicyRuleSetVersion = validation.StringInSlic
 var ValidateWebApplicationFirewallPolicyRuleSetType = validation.StringInSlice([]string{
```

Comment:
Hi Xu, this validation function is also used by `azurerm_application_gateway`, however the field in `azurerm_application_gateway` don't support the new value, with below error:
```
│ Error: creating/updating "Resource: (ResourceId \"/subscriptions/xxx/resourceGroups/wt-acctestRG/providers/Microsoft.Network/applicationGateways/testapgw\" / Api Version \"2023-06-01\")": PUT https://management.azure.com/subscriptions/xxx/resourceGroups/wt-acctestRG/providers/Microsoft.Network/applicationGateways/testapgw
│ --------------------------------------------------------------------------------
│ RESPONSE 400: 400 Bad Request
│ ERROR CODE: ApplicationGatewayFirewallUnknownRuleSetType
│ --------------------------------------------------------------------------------
│ {
│   "error": {
│     "code": "ApplicationGatewayFirewallUnknownRuleSetType",
│     "message": "RuleSetType 'Microsoft_DefaultRuleSet' does not exist for Application Gateway Firewall in context 'properties.webApplicationFirewallConfiguration'.",
│     "details": []
│   }
│ }
│ --------------------------------------------------------------------------------
│
```
So this validate function may need to be split. More details can be found in my duplicated PR https://github.com/hashicorp/terraform-provider-azurerm/pull/23899

Reply:
Seems will not be supported even in latest API `2023-06-01` as I have tested using azapi.